repm/types.go

```diff
@@ -83,6 +82,15 @@ type beginSyncResponse struct {
 	PullError *syncError  `json:"pullError,omitempty"`
 }
 
+type maybeEndSyncRequest struct {
+	SyncHead *jsnoms.Hash `json:"syncHead,omitempty"`
+}
+
+type maybeEndSyncResponse struct {
+	Ended           bool          `json:"ended,omitempty"`
```

Comment:
[minifix] The `Ended` field is non-DRY because it is the same information as `len(ReplayMutations) > 0`. It should be removed.

Reply:
@arv ^ ignore the ended field, i will remove it in a pr in the next day or so

Reply:
actually hol up. we get zero mutations on error and there is currently no way to distinguish as a caller of the repm api whether we got an error or we're done if we remove ended. (or, even that an error occurred at all). guess our choices are to have a success bool (what we have here) or to have an error field. the predominant style seems to be an error field so let's go with that.

obvious choices are to add `Internal string` to `syncError` or an `Error string` to the response. i'm not a fan of the `syncError` pattern in the code, i far prefer something like this embedded in each response:
```
type rpcResult struct {
  Code ResultCode
  Message string
}

const (
	ResultCodeOK = iota
	ResultCodeAuthError
        ResultCodeInternalError
)
```
Opinions?

Reply:
I thought that we return the error to the caller of repm here:

https://github.com/rocicorp/replicache-client/blob/master/repm/api.go#L208

So if bindings receive a non-error response to `MaybeEndSync()` that has zero mutations that need replaying, then it's done. What am I missing?

Reply:
More like what am I missing. It never occurred to me that we'd return errors in this go style out of the api. I guess I've been assuming (without even bothering to check) that since this api would serve many different bindings that the request and response contained the entirety of the information into and out of the api. 

I don't think we should necessarily change it, just saying the missing thing is a (probably not uncommon) assumption I made.

Reply:
Ah. Maybe that would make sense. Gomobile understands the normal Go convention so I just used it. I guess using the response struct would enable more detailed structured errors which could be good. OTOH, most languages do separate the concept of an error response from the success response, so that maps well.

Reply:
ok i will remove the bool and we can rely on empty replaymutations and no error to signal success. let's take an option on doing a more regular and explicit result in the future along the lines of the result/resultcode above.